plugins/adr/src/components/AdrReader/AdrReader.tsx

```diff
@@ -77,9 +83,26 @@ export const AdrReader = (props: {
         <WarningPanel title="Failed to fetch ADR" message={error?.message} />
       )}
 
-      {!loading && !error && value?.data && (
-        <MarkdownContent content={adrContent} linkTarget="_blank" />
+      {!backendUrlLoading && backendUrlError && (
+        <WarningPanel
+          title="Failed to fetch ADR images"
+          message={backendUrlError?.message}
+        />
       )}
+
+      {!loading &&
+        !backendUrlLoading &&
+        !error &&
+        !backendUrlError &&
+        value?.data && (
+          <MarkdownContent
+            content={adrContent}
+            linkTarget="_blank"
+            transformImageUri={href => {
+              return `${backendUrl}/image?url=${href}`;
```

Comment:
This will not work if authentication is enabled. Do we have the option to provide a custom image fetcher instead? so that we don't need to treat them as static resources but instead use a blob/data URI on the frontend?

Reply:
Hey, do you mean if it is possible to fetch the images on the frontend and not using backend at all?